### PR TITLE
mpidig miscellaneous cleanup

### PR DESCRIPTION
**Pull request by gcongiu**
## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [x] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by gcongiu** [comment]
test:jenkins/ch4/most

File: src/mpid/ch4/generic/am/mpidig_am.h [hunk 1]
```diff
@@ -36,4 +36,6 @@ int MPIDIG_am_reg_cb_dynamic(MPIDIG_am_origin_cb origin_cb, MPIDIG_am_target_msg
 int MPIDIG_am_init(void);
 void MPIDIG_am_finalize(void);
 
+int MPIDIG_comm_abort(MPIR_Comm * comm, int exit_code);
```
Comment (hzhou) [inline src/mpid/ch4/generic/am/mpidig_am.h]: If we are going to make the move, we need rename it to `MPIDIG_am_comm_abort`

Comment (gcongiu) [reply src/mpid/ch4/generic/am/mpidig_am.h]: Right, I will do that change last

Comment (gcongiu) [reply src/mpid/ch4/generic/am/mpidig_am.h]: I am still unsure whether this should actually be in a `common` dir inside `generic`. This is not `am` specific.

Comment (hzhou) [reply src/mpid/ch4/generic/am/mpidig_am.h]: It is `am` specific. It is falling back to a generic routine using `am`. In principle, it is equally valid to falling back to a generic routine using `gpu`.

Comment (gcongiu) [reply src/mpid/ch4/generic/am/mpidig_am.h]: Yes it is, sorry for the confusion.

**Comment by gcongiu** [comment]
test:jenkins/ch4/most

**Comment by gcongiu** [comment]
test:jenkins/ch4/ucx

**Comment by gcongiu** [comment]
test:mpich/warnings

**Comment by hzhou** [comment]
@gcongiu Merged #4311, now you have conflicts :)

**Comment by gcongiu** [comment]
test:mpich/warnings
test:jenkins/ch4/most

After you #4311 was merged this PR is left skin and bones :) Rerunning the tests

**Pull request merged**